cognite/client/_api/simulators/runs.py

```diff
@@ -217,3 +223,34 @@ def create(self, run: SimulationRunWrite | Sequence[SimulationRunWrite]) -> Simu
             input_resource_cls=SimulationRunWrite,
             resource_path=self._RESOURCE_PATH_RUN,
         )
+
+    def list_run_data(
+        self,
+        run_id: int,
+    ) -> SimulatorRunDataList:
+        """`Get simulation run data <https://developer.cognite.com/api#tag/Simulation-Runs/operation/simulation_data_by_run_id_simulators_runs_data_list_post>`_
+        Retrieve data associated with a simulation run by ID.
+
+        Args:
+            run_id (int): Simulation run id.
+
+        Returns:
+            SimulatorRunDataList: List of simulation run data
+
+        Examples:
+            Get simulation run data:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.runs.list_run_data(run_id=12345)
+        """
+        self._warning.warn()
+
+        req = self._post(
+            url_path=f"{self._RESOURCE_PATH}/data/list",
+            json={"items": [{"runId": run_id}]},
+        )
+
+        items = req.json().get("items", [])
+        run_items = [SimulationRunDataItem.load(item) for item in items]
+
+        return SimulatorRunDataList(resources=run_items, cognite_client=self._cognite_client)
```

Comment:
This can be simplified. Note that we use the private `_load` method. This is a tiny optimization as we know we have a dict and not a YAML and JSON string.
```suggestion
        return SimulatorRunDataList._load(items, cognite_client=self._cognite_client)
```